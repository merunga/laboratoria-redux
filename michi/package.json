{
  "name": "michi",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "prop-types": "^15.6.1",
    "react": "^16.2.0",
    "react-dom": "^16.2.0",
    "react-redux": "^5.0.7",
    "react-scripts": "1.1.1",
    "redux": "^3.7.2"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject",
    "lint": "eslint --fix ."
  },
  "devDependencies": {
    "eslint": "^4.19.0",
    "eslint-config-airbnb": "^16.1.0",
    "eslint-plugin-import": "^2.9.0",
    "eslint-plugin-jest": "^21.15.0",
    "eslint-plugin-jsx-a11y": "^6.0.3",
    "eslint-plugin-react": "^7.7.0"
  }
}
