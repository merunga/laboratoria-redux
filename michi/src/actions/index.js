export const actionTypes = {
  SELECT: 'SELECT',
  RESET: 'RESET',
}

export const select = (x, y, turno) => ({
  type: actionTypes.SELECT,
  x,
  y,
  turno,
})

export const reset = () => ({
  type: actionTypes.RESET,
})
