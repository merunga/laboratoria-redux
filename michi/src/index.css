* {
  margin: 0;
  padding: 0;
  font-family: 'VT323', monospace;
  text-transform: uppercase;
}
body, #root {
  height: 100vh;
}
#root {
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #000;
  color: lime;
  font-size: 30px;
}
button {
  background: #333;
  color: lime;
  border: 1px solid #444;
  padding: 7px 10px;
  font-size: 20px;
}