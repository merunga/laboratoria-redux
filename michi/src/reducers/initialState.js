export default {
  michi: [
    [null, null, null],
    [null, null, null],
    [null, null, null],
  ],
  // historia: {
  //   comienza: 'X',
  //   jugadas: [],
  // },
}
