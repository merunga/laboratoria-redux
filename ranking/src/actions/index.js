export const actionTypes = {
  UP_VOTE: 'UP_VOTE',
  DOWN_VOTE: 'DOWN_VOTE',
  PREV: 'PREV',
  NEXT: 'NEXT',
}

export const upVote = id => ({
  type: actionTypes.UP_VOTE,
  id,
})

export const downVote = id => ({
  type: actionTypes.DOWN_VOTE,
  id,
})

export const prev = () => ({
  type: actionTypes.PREV,
})

export const next = () => ({
  type: actionTypes.NEXT,
})
