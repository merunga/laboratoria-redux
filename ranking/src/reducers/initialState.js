export default {
  canciones: [
    {
      id: 1, titulo: 'Psycho', autor: 'Post Malone', votos: 70,
    },
    {
      id: 2, titulo: 'Perfect', autor: 'Ed Sheeran', votos: 90,
    },
    {
      id: 3, titulo: 'God\'s Plan', autor: 'Drake', votos: 100,
    },
    {
      id: 4, titulo: 'Finesse', autor: 'Bruno Mars', votos: 80,
    },
    {
      id: 5, titulo: 'Meant To Be', autor: 'Bebe Rexha & Florida Georgia Line', votos: 60,
    },
  ],
  seleccionado: 1,
}
