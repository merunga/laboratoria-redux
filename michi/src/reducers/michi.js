import { actionTypes } from '../actions'

export default (state, action) => state || null
