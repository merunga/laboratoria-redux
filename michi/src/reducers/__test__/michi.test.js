import initialState from '../initialState'
import michi from '../michi'
import * as actions from '../../actions'

describe('michi', () => {
  it('seleccion', () => {
    const row = [0, 1, 2]
    /* eslint-disable array-callback-return */
    row.map((i) => {
      row.map((j) => {
        expect(michi(initialState.michi, actions.select(i, j, 'X'))).toMatchSnapshot()
      })
    })
    /* eslint-enable array-callback-return */
  })
  it('reset', () => {
    expect(michi(initialState.michi, actions.reset())).toMatchSnapshot()
  })
})
