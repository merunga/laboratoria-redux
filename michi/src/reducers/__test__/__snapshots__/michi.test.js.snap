// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`michi reset 1`] = `
Array [
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
]
`;

exports[`michi seleccion 1`] = `
Array [
  Array [
    "X",
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
]
`;

exports[`michi seleccion 2`] = `
Array [
  Array [
    null,
    "X",
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
]
`;

exports[`michi seleccion 3`] = `
Array [
  Array [
    null,
    null,
    "X",
  ],
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
]
`;

exports[`michi seleccion 4`] = `
Array [
  Array [
    null,
    null,
    null,
  ],
  Array [
    "X",
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
]
`;

exports[`michi seleccion 5`] = `
Array [
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    "X",
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
]
`;

exports[`michi seleccion 6`] = `
Array [
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    "X",
  ],
  Array [
    null,
    null,
    null,
  ],
]
`;

exports[`michi seleccion 7`] = `
Array [
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
  Array [
    "X",
    null,
    null,
  ],
]
`;

exports[`michi seleccion 8`] = `
Array [
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    "X",
    null,
  ],
]
`;

exports[`michi seleccion 9`] = `
Array [
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    null,
  ],
  Array [
    null,
    null,
    "X",
  ],
]
`;
